src/cpp/database/database.cpp

```diff
@@ -2114,6 +2120,577 @@ EntityStatus Database::get_entity_status(
+            break;
+        }
+        case EntityKind::TOPIC:
+        {
+            update_endpoint_in_graph(domain_id, EntityId(), entity_id, EntityId());
+            break;
+        }
+        case EntityKind::DATAREADER:
+        case EntityKind::DATAWRITER:
+        {
+            std::shared_ptr<const database::DDSEndpoint> endpoint =
+                    std::dynamic_pointer_cast<const database::DDSEndpoint>(entity);
+            if (endpoint->participant == nullptr)
+            {
+                return;
+            }
+            std::shared_ptr<const database::DomainParticipant> participant = endpoint->participant;
+            update_endpoint_in_graph(domain_id, participant->id, EntityId(), entity_id);
+            break;
+        }
+        default:
+        {
+            break;
+        }
+
+            if (graph_updated)
+            {
+                details::StatisticsBackendData::get_instance()->on_domain_graph_update(domain_id);
+            }
+    }
+}
+
+Graph Database::get_entity_subgraph(
+        const EntityId& entity_id,
+        Graph& entity_graph)
+{
+    bool entity_graph_updated = false;
+
+    std::shared_ptr<const database::Entity> entity = get_entity(entity_id);
+
+    entity_graph["kind"] =  entity_kind_str[(int)entity->kind];
+
+    if (entity_graph["alias"] != entity->alias)
+    {
+        entity_graph["alias"] =  entity->alias;
+        entity_graph_updated = true;
+    }
+
+    entity_graph["metatraffic"] =  entity->metatraffic;
+
+    if (entity->kind != EntityKind::TOPIC && entity_graph["status"] != entity_status_str[(int)entity->status])
+    {
+        entity_graph["status"] = entity_status_str[(int)entity->status];
+        entity_graph_updated = true;
+    }
+
+    if (entity->kind == EntityKind::PROCESS)
+    {
+        std::shared_ptr<const database::Process> process =
+                std::dynamic_pointer_cast<const database::Process>(entity);
+        entity_graph["pid"] =  process->pid;
+    }
+
+    if (entity->kind == EntityKind::PARTICIPANT)
+    {
+        std::shared_ptr<const database::DomainParticipant> participant =
+                std::dynamic_pointer_cast<const database::DomainParticipant>(entity);
+        if (entity_graph["app_id"] != app_id_str[(int)participant->app_id])
+        {
+            entity_graph["app_id"] =  app_id_str[(int)participant->app_id];
+            entity_graph_updated = true;
+        }
+        if (entity_graph["app_metadata"] != participant->app_metadata)
+        {
+            entity_graph["app_metadata"] =  participant->app_metadata;
+            entity_graph_updated = true;
+        }
+    }
+
+    if (entity->kind == EntityKind::DATAWRITER ||
+            entity->kind == EntityKind::DATAREADER)
+    {
+        std::shared_ptr<const database::DDSEndpoint> endpoint =
+                std::dynamic_pointer_cast<const database::DDSEndpoint>(entity);
+        entity_graph["topic"] =  std::to_string(endpoint->topic->id.value());
+    }
```

Comment:
Why not use a `switch` statement as the previous function?

Reply:
Modified in PR #207